### ISSUE TITLE
__libc_init_array() crash (newlib)

### ISSUE THREAD

**Issue by dylad** [RIOT-OS/RIOT#9248]
#### Description

I'm encountering a weird issue with the __libc_init_array(); called in cpu/cortexm_common/vectors_cortexm.c
I tried on saml21-xpro, nrf52dk and with tests/leds app. My LEDs are not blinking :(
I'm using Ubuntu 18.04

I guess I did something wrong when I setup my ubuntu ?...

#### Steps to reproduce the issue

Use lastest ubuntu 18.04 iso
install all dependencies for build, etc.
#### Expected results

LEDs blink
#### Actual results

MCU hangs
#### Versions

Ubuntu 18.04 on VM
```
./dist/tools/ci/print_toolchain_versions.sh 
Installed compiler toolchains 
-----------------------------
             native gcc: gcc (Ubuntu 7.3.0-16ubuntu3) 7.3.0
      arm-none-eabi-gcc: arm-none-eabi-gcc (15:6.3.1+svn253039-1build1) 6.3.1 20170620
                avr-gcc: missing
       mips-mti-elf-gcc: missing
             msp430-gcc: missing
   riscv-none-embed-gcc: missing
                  clang: missing

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: "2.4.0"
    mips-mti-elf-newlib: missing
riscv-none-embed-newlib: missing
               avr-libc: missing (missing)

Installed development tools
---------------------------
                  cmake: cmake version 3.10.2
               cppcheck: missing
                doxygen: missing
                 flake8: missing
                    git: git version 2.17.0
             coccinelle: missing

```

**Comment by adjih** [comment]
I suspect I met the same problem: 

- using ubuntu 18.04, with base packages, tried to flash hello-world (master branch of RIOT) 

- -> nothing was output.

 I was advised to use arm-gcc 7 instead; doing so: 

- -> Hello-world worked again.

**Comment by dylad** [comment]
Good to know I'm not the only one !
Maybe there's nothing with gcc-arm-none-eabi version of ubuntu 18.04 ?

**Comment by kb2ma** [comment]
I experienced a similar problem in #9846, and the approach described in this issue fixed the problem. Rather than just bring in arm-gcc 7, I replaced the tools completely via Ubuntu packages, using instructions from the Launchpad gcc arm embedded toolchain maintainers [1]. 

1. Add the team-gcc-arm-embedded repository to apt.
1. Remove packages: gcc-arm-none-eabi, binutils-arm-none-eabi, libnewlib-arm-none-eabi, libstdc++-arm-none-eabi-newlib
1. Also remove AVR gcc packages due to a file conflict: gcc-avr, avr-libc, binutils-avr. This seems like a bug, so I plan to notify them.
1. Install the new package: gcc-arm-embedded. At time of writing, the package is at revision 7-2018qs-1~bionic1

[1] https://launchpad.net/~team-gcc-arm-embedded/+archive/ubuntu/ppa

**Comment by dylad** [comment]
@kb2ma I'm glad this solve your issue !
I'm also closing this issue since this is outside the scope of RIOT-OS.

**Comment by cladmi** [comment]
@kb2ma @dylad do you know a way to detect it is the ubuntu bionic version ?
(except doing `dpkg-query` because it is really slow…)

As it is a problem that will come back again it would be great if we can trigger a warning when building.

**Comment by kb2ma** [comment]
How about [CheckingYourUbuntuVersion](https://help.ubuntu.com/community/CheckingYourUbuntuVersion)? See the terminal section of the page.

**Comment by cladmi** [comment]
Sorry, I meant the arm-gcc version installed from ubuntu bionic.

**Comment by gebart** [comment]
@cladmi I guess arm-none-eabi-gcc -version will show some particular version for this broken toolchain

**Comment by kb2ma** [comment]
Whoops, should have read more context. :-) Would apt-cache work? I see this with the new package.

> $ apt-cache show gcc-arm-embedded |grep Version
> Version: 7-2018q2-1~bionic1

**Comment by cladmi** [comment]
See here https://github.com/RIOT-OS/RIOT/pull/10404

**Comment by kb2ma** [comment]
> Rather than just bring in arm-gcc 7, I replaced the tools completely via Ubuntu packages...

FWIW, the package issue looks to be resolved in Ubuntu 19.04. It now uses version 7-2081-q2-6 of the gcc-arm-none-eabi package. It works OK for me.
